psycopg3/psycopg3/cursor.py

```diff
@@ -43,13 +44,17 @@
     execute = generators.execute
 
 
+def default_row_factory(cursor: Any) -> RowMaker:
```

Comment:
Not sure how much it gets in the way with the type system, or how much a micro-optimisation it is, but I would rather avoid a Python call per row in case the client doesn't set a row factory. So maybe I wouldn't have a no-op default row-maker but rather a None and a `return self._make_row(t) if self._make_row else t` sort of thing.

Reply:
Done in commit "Make row factory optional" I just pushed; and dropped the `default_row_factory()` accordingly. This turns out to be `return self._make_row(t) if self._make_row else cast(Row, t)` in order to pass type checking but maybe there's another solution?

Reply:
Not that I know. I tried dabbling with a factory_function cheating and returning None instead of a no-op callable but got nowhere.

If we cannot make mypy happy, while I'm ok to use `cast()`, probably a `# type: ignore` is good enough.

I was thinking just now that maybe it would be good to add a `row_factory` attribute to the `Transformer`, so that `load_row()` and `load_rows()` may return directly a `Row`? I think it would require to intervene in way less code paths. Or to give the transformer just a `RowMaker`.

Reply:
...The advantage of giving it a RowMaker is that we would still be able to create it in the cursor with a `self._tx.row_maker = self.row_factory(self)` at the right point.

Reply:
I took my a bit of work, but I finally made this idea working. It's indeed cleaner (no change in cursor code, besides type annotations).

Reply:
Yes thank you: it does seem cleaner indeed! Now the row_maker is used in only two places and other code paths (e.g. copy) should already be able to make use of it. Looks good!